### ISSUE TITLE
版本2.1.6.0, 将uuid(去掉“-”)作为nchar类型的tag, 以超表为基准对此tag做where条件查询无效

### ISSUE THREAD

**Issue by shadowycgg** [taosdata/TDengine#7488]
**1. 首先创建超级表:**
> create stable if not exists telemetry (ts TIMESTAMP, value DOUBLE) TAGS (entity_id NCHAR(255), entity_type NCHAR(255), key_name NCHAR(255));

> Query OK, 0 of 0 row(s) in database (0.006950s)

**2. 插入一条自动创建子表的数据:**
    
>insert into d8b0fc9f89c4f4ce6b95565f076853399 using telemetry tags('8b0fc9f89c4f4ce6b95565f076853399', 'DEVICE','温度') values(now, 2.33); 

> Query OK, 1 of 1 row(s) in database (0.022771s)

**3. 直接查询超表:**

> select * from telemetry;

>  ts            |           value           |           entity_id            |          entity_type           |            key_name            |
=========================================================================================================================================================
 2021-08-19 22:40:58.253 |               2.330000000 | 8b0fc9f89c4f4ce6b95565f0768... | DEVICE                         | 温度                           |
Query OK, 1 row(s) in set (0.037708s)

**4.按超级表tag进行where条件查询, 无法得出正确结果, 返回数据条目数为0:**

>  select * from telemetry where entity_id = '8b0fc9f89c4f4ce6b95565f076853399';

> Query OK, 0 row(s) in set (0.005878s) 

**5.切换查询条件, 查询结果正确:**

> select * from telemetry where entity_type = 'DEVICE';

>   ts            |           value           |           entity_id            |          entity_type           |            key_name            |
=========================================================================================================================================================
 2021-08-19 22:40:58.253 |               2.330000000 | 8b0fc9f89c4f4ce6b95565f0768... | DEVICE                         | 温度                           |
Query OK, 1 row(s) in set (0.051216s)

**6.将uuid取前20位, 插入一条数据**

> insert into d8b0fc9f89c4f4ce6b955 using telemetry tags('8b0fc9f89c4f4ce6b955', 'DEVICE','温度') values(now, 2.33);

> Query OK, 1 of 1 row(s) in database (0.020156s)

**7.where条件查询结果正确:**

> select * from telemetry where entity_id = '8b0fc9f89c4f4ce6b955';

>  ts            |           value           |           entity_id            |          entity_type           |            key_name            |
=========================================================================================================================================================
 2021-08-19 22:55:54.669 |               2.330000000 | 8b0fc9f89c4f4ce6b955           | DEVICE                         | 温度                           |
Query OK, 1 row(s) in set (0.050629s)

**Comment by jtcheng** [comment]
Works well in version 2.1.7.0.
```
taos> select server_version();
 server_version() |
===================
 2.1.7.0          |
Query OK, 1 row(s) in set (0.000241s)

taos> select * from telemetry where entity_id = '8b0fc9f89c4f4ce6b95565f076853399';
           ts            |           value           |           entity_id            |          entity_type           |            key_name            |
=========================================================================================================================================================
 2021-08-20 00:04:11.944 |               2.330000000 | 8b0fc9f89c4f4ce6b95565f0768... | DEVICE                         | 温度                           |
Query OK, 1 row(s) in set (0.005057s)

taos>
```

**Comment by yu285** [comment]
yes，we fix this problem in 2.1.7
